tensorboard/program.py

```diff
@@ -341,79 +342,128 @@ def __init__(self, msg):
+        try:
+          self._server = cls(wsgi_app=wsgi_app, flags=subflags)
+          break
+        except TensorBoardFailedToBindError:
+          if not should_scan:
+            raise
+      else:
+        # All attempts failed to bind.
+        raise TensorBoardServerException(
+            'TensorBoard could not bind to any port around %s '
+            '(tried %d times)'
+            % (base_port, max_attempts))
+
+    def serve_forever(self):
+      return self._server.serve_forever()
+
+    def get_url(self):
+      return self._server.get_url()
+
+  return PortScanningServer
+
+
+@with_port_scanning
 class WerkzeugServer(serving.ThreadedWSGIServer, TensorBoardServer):
   """Implementation of TensorBoardServer using the Werkzeug dev server."""
+
   # ThreadedWSGIServer handles this in werkzeug 0.12+ but we allow 0.11.x.
   daemon_threads = True
 
   def __init__(self, wsgi_app, flags):
     self._flags = flags
     host = flags.host
-
-    # base_port: what's the first port to which we should try to bind?
-    # should_scan: if that fails, shall we try additional ports?
-    # max_attempts: how many ports shall we try?
-    should_scan = flags.port is None
-    base_port = core_plugin.DEFAULT_PORT if flags.port is None else flags.port
-    max_attempts = 10 if should_scan else 1
-
-    if base_port > 0xFFFF:
-      raise TensorBoardServerException(
-          'TensorBoard cannot bind to port %d > %d' % (base_port, 0xFFFF)
-      )
-    max_attempts = 10 if should_scan else 1
-    base_port = min(base_port + max_attempts, 0x10000) - max_attempts
+    port = flags.port
 
     # Without an explicit host, we default to serving on all interfaces,
     # and will attempt to serve both IPv4 and IPv6 traffic through one
     # socket.
     self._auto_wildcard = not host
+    if self._auto_wildcard:
+      host = self._get_wildcard_address(port)
 
-    for (attempt_index, port) in (
-        enumerate(xrange(base_port, base_port + max_attempts))):
-      if self._auto_wildcard:
-        host = self._get_wildcard_address(port)
-      try:
-        # Yes, this invokes the super initializer potentially many
-        # times. This seems to work fine, and looking at the superclass
-        # chain (type(self).__mro__) it doesn't seem that anything
-        # _should_ go wrong (nor does any superclass provide a facility
-        # to do this natively).
-        #
-        # TODO(@wchargin): Refactor so that this is no longer necessary,
-        # probably by switching composing around a Werkzeug server
-        # rather than inheriting from it.
-        super(WerkzeugServer, self).__init__(host, port, wsgi_app)
-        break
-      except socket.error as e:
-        if hasattr(errno, 'EACCES') and e.errno == errno.EACCES:
-          raise TensorBoardServerException(
-              'TensorBoard must be run as superuser to bind to port %d' %
+    try:
+      serving.ThreadedWSGIServer.__init__(self, host, port, wsgi_app)
```

Comment:
Comment for why this isn't a call to super()?

Reply:
No longer needed.